lib/Ast.ml

```diff
@@ -2384,7 +2510,28 @@ end = struct
     | Exp {pexp_desc= Pexp_field (e, _); _}, {pexp_desc= Pexp_construct _; _}
       when e == exp ->
         true
-    | Exp {pexp_desc; _}, _ -> (
+    | Exp ({pexp_desc; _} as ctx_exp), _ -> (
+      (* JST: This is the old fallthrough case, but we need it for the
```

Comment:
Oh, that's a habit I adopted inconsistently for comments in `ocaml-jst` that were about the code differences; there, at least, the `JST` corresponds with the `-jst` in the name.  Less obvious here!  I think it's sort of intended to signify "here's some code that's just here to support being different", rather than "here's a change that could be upstreamed".  I think it's worth calling out, but that may not be the best prefix, yeah.